src/features/activity/ActivityDetailsScreen.tsx

```diff
@@ -0,0 +1,297 @@
+    )
+  }, [colors, transaction])
+
+  const onAddressItemPress = useCallback(
+    (address: string) => () => {
+      setSelectedAddress(address)
+      setOptionsOpen(true)
+    },
+    [],
+  )
+
+  const AccountAddressListItems = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    if (confirmedSig.err || !enrichedTx.type) {
+      return null
+    }
+
+    const firstTokenTransfer = enrichedTx.tokenTransfers[0]
+    const firstNativeTransfer = enrichedTx.nativeTransfers[0]
+    const fromAccount =
+      firstTokenTransfer?.fromUserAccount ||
+      firstNativeTransfer?.fromUserAccount
+    const toAccount =
+      firstTokenTransfer?.toUserAccount || firstNativeTransfer?.toUserAccount
+
+    return (
+      <Box marginTop="l" flex={1} width="100%">
+        {fromAccount && (
+          <AddressActivityItem
+            accountAddress={fromAccount}
+            marginHorizontal="l"
+            marginBottom="xs"
+            borderRadius="xl"
+            showBubbleArrow
+            onMenuPress={onAddressItemPress(fromAccount)}
+          />
+        )}
+        {toAccount && (
+          <AddressActivityItem
+            accountAddress={toAccount}
+            marginHorizontal="l"
+            borderRadius="xl"
+            onMenuPress={onAddressItemPress(toAccount)}
+          />
+        )}
+      </Box>
+    )
+  }, [onAddressItemPress, transaction])
+
+  const title = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    if (confirmedSig.err) {
+      return t('activityScreen.transactionFailed')
+    }
+
+    const txnKey = `activityScreen.enrichedTransactionTypes.${enrichedTx.type}`
+    return i18n.exists(txnKey)
+      ? t(`activityScreen.enrichedTransactionTypes.${enrichedTx.type}`)
+      : t('activityScreen.enrichedTransactionTypes.UNKNOWN')
+  }, [i18n, t, transaction])
+
+  const description = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    // Custom description that elipiizes the address
+    if (enrichedTx.description) {
+      const customDescription = enrichedTx.description
+        ?.split(' ')
+        .map((word) => {
+          // Remove addresses
+          if (solAddressIsValid(word)) {
+            return ''
+          }
+          return word
+        })
+        .join(' ')
+        .trim()
+
+      // capitalize first letter of description
+      return `${customDescription
+        .charAt(0)
+        .toUpperCase()}${customDescription.slice(1)}`
+    }
+
+    if (confirmedSig.err) {
+      return 'Error'
+    }
+
+    return 'Transaction Successful'
+  }, [transaction])
+
+  const handleOpenExplorer = useCallback(async () => {
+    const url = `https://explorer.solana.com/tx/${transaction.signature}?cluster=${cluster}`
```

Comment:
I'd rather keep these urls in one place. Could you use `useCreateExplorerUrl` here? You'll probably need to update the hook to support different clusters.

```ts
const createExplorerUrl = useCreateExplorerUrl()

const handleOpenExplorer = useCallback(async () => {
  const url = createExplorerUrl('txn', transaction.signature, {cluster})
  await Linking.openURL(url)
  }, [cluster, transaction.signature])

Reply:
Will do. Didn't even know we had hat 😅

Reply:
Done